airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/write/object_storage/ObjectLoader.kt

```diff
@@ -53,6 +54,19 @@ import io.airbyte.cdk.load.write.LoadStrategy
  *
  * The parts are distributed evenly across the loading and completing workers w/o regard to object
  * key or stream. This is currently not configurable, but tests have shown it is optimal.
+ *
+ * Extending for use in other interfaces:
+ *
+ * [ObjectLoader] is designed to compose with other interfaces that use object loading as a first
+ * step. To do so, provide an `@Named("objectLoaderCompletedUploadQueue")` queue of type
+ * [io.airbyte.cdk.load.message.PartitionedQueue] for the output and extend
+ * [io.airbyte.cdk.load.pipeline.object_storage.ObjectLoaderCompletedUploadPartitioner] to partition
+ * it. (See [io.airbyte.cdk.load.pipeline.db.BulkLoadCompletedUploadQueue] and
+ * [io.airbyte.cdk.load.pipeline.db.BulkLoadCompletedUploadPartitioner] for an example.
+ * Additionally, CDK devs building new interfaces will probably want to set `[stateAfterUpload] to
```

Comment:
```suggestion
 * Additionally, CDK devs building new interfaces should set `[stateAfterUpload] to
```

(I don't think there's _ever_ a case where you want to return COMPLETE, and then run another step afterward?)